### PR TITLE
DRAFT: Wraps tests in IIFE.

### LINKED ISSUE #15790: Scrollable View tests are broken
The latest VS Code merge broke these tests so I disabled them for now 

https://github.com/microsoft/azuredatastudio/blob/main/src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts#L49

The issue seemed to be because the tests rely on animation frames being ran, but when running the unit tests the UI is hidden and so the animation frames don't seem to be triggered and thus the test just times out.

We should either fix them or remove them if they don't seem useful enough to keep around.

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR fixes #15790 

The changes included in this PR are to fix our broken unit tests around scrollable views.

### CODE REVIEW

File: src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts [hunk 1]
```diff
@@ -63,128 +63,141 @@ suite.skip('ScrollableView', () => { // TODO chgagnon Fix these tests
 		scrollableView.dispose();
 	});
 
-	test('adds and removes views correctly', async () => {
-		const view1 = new TestView(20, 20);
-		const view2 = new TestView(20, 20);
-		const view3 = new TestView(20, 20);
-		const scrollableView = new ScrollableView(container);
+	test('adds and removes views correctly', async (): Promise<void> => {
+		(async () => {
```
Comment (Charles-Gagnon) [inline src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts:67]: I don't think this is going to help - unless I'm mistaken these tests will pass but that's because they're essentially not running. By wrapping them in an async function that you don't await on the test will immediately return and thus "pass" even if the wrapped test fails. 

What were you trying to accomplish with this approach?

File: src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts [hunk 2]
```diff
@@ -63,128 +63,141 @@ suite.skip('ScrollableView', () => { // TODO chgagnon Fix these tests
...
+			scrollableView.addView(view1);
 
-		await waitForAnimation();
+			await waitForAnimation();
 
-		assert.equal(view1.size, 200, 'view1 is entire size');
+			assert.equal(view1.size, 200, 'view1 is entire size');
 
-		scrollableView.addView(view2);
+			await waitForAnimation();
```
Comment (Charles-Gagnon) [inline src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts:113]: Why did you remove the addView call?

File: src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts:67]: Huh, or maybe I'm wrong? The tests do appear to be running and failing in the CI build...which is definitely not what I would have expected. Do you have context here on what's making that work?

File: src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts (hunk 2 above)
Comment (lewis-sanchez) [reply src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts:113]: Sorry about that, I must have lost that line when I was tinkering with the surrounding async calls. I'll add it back.

File: src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/base/test/browser/ui/scrollableView/scrollableView.test.ts:67]: Looking at it closer - the errors are being caught by the unhandled error check (`should not have unexpected errors:`)

So what I said does look like it's happening. The tests are passing because they kick off an async function and immediately return 

image

I'm going to make an educated guess that the tests are passing without errors on Linux because we set up a virtual frame buffer to simulate a screen (https://github.com/microsoft/azuredatastudio/blob/main/build/azure-pipelines/linux/sql-product-build-linux.yml#L116). Windows & Mac don't have any such thing set up and the lab machines are run without a UI context and so the animation stuff is never triggering. 

So either have to:

1. Figure out if we can do a similar thing for Mac/Windows
2. Figure out how we can get the animation stuff to tick even without an actual UI context available
